test_apps/advection/benchmark.cpp

```diff
@@ -0,0 +1,217 @@
+    }
+  }
+  return 0;
+}
+
+struct {
+  string datapath;
+  std::vector <string> fieldname;
+  int seeds;
+  int steps;
+  double length;
+  std::vector <double> minu;
+  std::vector <double> maxu;
+} opt;
+
+OptionParser::OptDescRec_T  set_opts[] = {
+  {"datapath", 1,  "", "Path to the parent directory of the dataset"},
+  {
+    "fieldname",  1,  "",  "Colon delimited 3-element vector "
+    "specifying field names for X, Y, Z components (u:v:w)"
+  },
+  {"seeds", 1,  "100", "Number of random seeds for the benchmark"},
+  {"steps", 1,  "100", "Maximum number of advection steps for particles"},
+  {"length", 1,  "", "Prescribed step length for an advection step"},
+  {
+    "minu",  1,  "",  "Colon delimited 3-element vector "
+    "specifying rake min extents in user coordinates (X0:Y0:Z0)"
+  },
+  {
+    "maxu",  1,  "",  "Colon delimited 3-element vector "
+    "specifying rake max extents in user coordinates (X1:Y1:Z1)"
+  },
+  {NULL}
+};
+
+OptionParser::Option_T  get_options[] = {
+  {"datapath", Wasp::CvtToCPPStr, &opt.datapath, sizeof(opt.datapath)},
+  {"fieldname", Wasp::CvtToStrVec, &opt.fieldname, sizeof(opt.fieldname)},
+  {"seeds", Wasp::CvtToInt, &opt.seeds, sizeof(opt.seeds)},
+  {"steps", Wasp::CvtToInt, &opt.steps, sizeof(opt.steps)},
+  {"length", Wasp::CvtToDouble, &opt.length, sizeof(opt.length)},
+  {"minu", Wasp::CvtToDoubleVec, &opt.minu, sizeof(opt.minu)},
+  {"maxu", Wasp::CvtToDoubleVec, &opt.maxu, sizeof(opt.maxu)},
+  {NULL}
+};
+
+const char  *ProgName;
+
+int main (int argc, char** argv)
+{
+
+  OptionParser op;
+
+  ProgName = FileUtils::LegacyBasename(argv[0]);
+
+  MyBase::SetErrMsgFilePtr(stderr);
+
+  if (op.AppendOptions(set_opts) < 0) {
+    cerr << ProgName << " : " << op.GetErrMsg();
+    exit(1);
+  }
+  if (op.ParseOptions(&argc, argv, get_options) < 0) {
+    cerr << ProgName << " : " << op.GetErrMsg();
+    exit(1);
+  }
+
+  const std::string datapath = opt.datapath;
+  // Field is not really used
+  const std::string fieldx = opt.fieldname.at(0);
+  const std::string fieldy = opt.fieldname.at(1);
+  const std::string fieldz = opt.fieldname.at(2);
+
+  const long numSeeds = opt.seeds;
+  const long steps = opt.steps;
+  float length = opt.length;
+#ifdef OMP_POSSIBLE
+  std::cout << "Available # of threads : " << omp_get_max_threads() << std::endl;
+#endif
+  std::cout << "Advection w/ : "
+            << "\nData : " << datapath
+            << "\nField : " << fieldx << "| " << fieldy << " | " << fieldz
+            << "\nSeeds : " << numSeeds
+            << "\nSteps : " << steps
+            << "\nLength : " << length  << std::endl;
+
+  int res;
+  // TODO : read field and variable information
+  // Let's assume we have read the data and field
+  const std::string filetype = "cf";
+  const size_t cache = 2000;
+  const size_t threads = 0;
+
+  std::vector<std::string> files;
+  files.push_back(datapath);
+  std::vector<std::string> fileopts;
+  //fileopts.push_back("-project_to_pcs");
```

Comment:
please remove commented out code

Reply:
I've left some comments which would make it easier to test the validity of the output, but mostly cleaned all of them.